commands/lsp_metals_show_build_target_info.py

```diff
@@ -0,0 +1,37 @@
+from . lsp_metals_text_command import LspMetalsTextCommand
+from . utils import handle_error
+from LSP.plugin.core.protocol import Error
+from LSP.plugin.core.types import List, Any
+
+import sublime
+
+
+class LspMetalsShowBuildTargetInfoCommand(LspMetalsTextCommand):
+    _command_name = 'list-build-targets'
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session:
+            params = {
+              "command": self._command_name
+            }
+            session.execute_command(params, progress=True).then(self._handle_response)
+
+    def _handle_response(self, response: Any) -> None:
+        if isinstance(response, Error):
+            handle_error(self._command_name, response)
+        elif isinstance(response, list) and response:
+            self.view.window().show_quick_panel(
+                response,
+                lambda i: self._on_select(response, i),
+                placeholder="Select the build target to display")
```

Comment:
You can override `handle_success_async` instead and then you don't need to define duplicated `run`. See implementation of `LspExecuteCommand`: https://github.com/sublimelsp/LSP/blob/566a49e2e4c8f9cfd057861681b0117bbe274bd2/plugin/execute_command.py#L49-L69

Reply:
Thanks, I think I can refactor other commands this way as well.